src/daft-micropartition/src/partitioning.rs

```diff
@@ -0,0 +1,208 @@
+use std::{collections::HashMap, sync::Arc};
+
+use common_error::{DaftError, DaftResult};
+use daft_dsl::Expr;
+use futures::stream::BoxStream;
+
+use crate::MicroPartition;
+type PartId = String;
```

Comment:
```suggestion
type PartitionId = String;
```

I like not have abbreviations

Reply:
most of the naming here was copied over from the python implementation

Reply:
hmmm ok 🥲 I guess it is fine... ideally we could rename both but I suppose that doesn't make this PR ⚛️